optuna/storages/_rdb/alembic/versions/v3.0.0.b.py

```diff
@@ -15,10 +24,45 @@
 branch_labels = None
 depends_on = None
 
+BaseModel = declarative_base()
 FLOAT_PRECISION = 53
 
 
+class TrialState(enum.Enum):
+    RUNNING = 0
+    COMPLETE = 1
+    PRUNED = 2
+    FAIL = 3
+    WAITING = 4
+
+
+class TrialModel(BaseModel):
+    __tablename__ = "trials"
+    trial_id = Column(Integer, primary_key=True)
+    number = Column(Integer)
+    state = Column(Enum(TrialState), nullable=False)
+
+
+class TrialValueModel(BaseModel):
+    __tablename__ = "trial_values"
+    trial_value_id = Column(Integer, primary_key=True)
+    trial_id = Column(Integer, ForeignKey("trials.trial_id"), nullable=False)
+    value = Column(Float, nullable=False)
+
+
 def upgrade():
+    bind = op.get_bind()
+    session = Session(bind=bind)
+
+    if (
+        session.query(TrialValueModel)
+        .join(TrialModel, TrialValueModel.trial_id == TrialModel.trial_id)
+        .filter(and_(TrialModel.state == TrialState.COMPLETE, TrialValueModel.value.is_(None)))
+        .count()
+    ) != 0:
+        raise ValueError("Found invalid trial_values' records (value=None and state='Complete')")
```

Comment:
To apply yanase-san's suggestion (See https://github.com/optuna/optuna/pull/3668/files#r901603363).

```suggestion
        raise ValueError("Found invalid trial_values records (value=None and state='COMPLETE')")
```